state/endpoint_bindings.go

```diff
@@ -269,18 +298,18 @@ func validateEndpointBindingsForMachines(a *Application, oldBindings, newBinding
 		if err != nil {
 			return errors.Annotatef(err, "unable to get space assignments for machine %q", m.Id())
 		}
-		for spName := range machineSpaces {
-			machineCountInSpace[spName]++
+		for spID := range machineSpaces {
+			machineCountInSpace[spID]++
 		}
 	}
 
-	if newDefaultSpace, defined := newBindings[""]; defined && newDefaultSpace != network.DefaultSpaceName {
+	if newDefaultSpace, defined := newBindings[defaultEndpointName]; defined && newDefaultSpace != network.DefaultSpaceId {
 		if machineCountInSpace[newDefaultSpace] != len(deployedMachines) {
 			return errors.Errorf("changing default space to %q is not feasible: one or more deployed machines lack an address in this space", newDefaultSpace)
```

Comment:
This error should include the space name

Reply:
updated